File: TO-DO_banckend/todo-management/src/main/java/net/javaguides/todo/service/TodoService.java
package net.javaguides.todo.service;

import net.javaguides.todo.dto.TodoDto;
import net.javaguides.todo.entity.Todo;

import java.util.List;

public interface TodoService {

    TodoDto addTodo(TodoDto todoDto);

    TodoDto getTodo(Long id);

    List<TodoDto> getAllTodos();

    TodoDto updateTodo(TodoDto todoDto, Long id);

    Void deleteTodo(Long id);

    TodoDto completeTodo(Long id);

    TodoDto inCompleteTodo(Long id);
}

File: TO-DO_frontend/todo-ui/src/services/TodoService.js
import axios from 'axios';

const TODO_API_BASE_URL = "http://localhost:8080/api/todos";

export const getAllTodos = () => axios.get(TODO_API_BASE_URL);

export const saveTodo = (todo) => axios.post(TODO_API_BASE_URL, todo);

export const getTodo = (id) => axios.get(TODO_API_BASE_URL + '/' + id);

export const updateTodo = (id, todo) => axios.put(TODO_API_BASE_URL + '/' + id, todo);

export const deleteTodo =(id) => axios.delete(TODO_API_BASE_URL + '/' + id);

export const completeTodo = (id) => axios.patch(TODO_API_BASE_URL + '/' + id + '/complete');

export const inCompleteTodo =(id) => axios.patch(TODO_API_BASE_URL + '/' + id + '/in-complete');